src/Xamarin.Android.Tools.AndroidSdk/AndroidSdkInfo.cs

```diff
@@ -155,12 +155,13 @@ internal static void DefaultConsoleLogger (TraceLevel level, string message)
 			}
 		}
 
-		public static void SetPreferredAndroidSdkPath (string path, Action<TraceLevel, string> logger = null)
+		public static bool SetPreferredAndroidSdkPath (string path, Action<TraceLevel, string> logger = null)
 		{
 			logger  = logger ?? DefaultConsoleLogger;
 
-			var sdk = CreateSdk (logger);
-			sdk.SetPreferredAndroidSdkPath (path);
+                        var sdk = CreateSdk(logger); 
+                        bool setSDKPath = sdk.SetPreferredAndroidSdkPath(path);
+                        return setSDKPath;
 		}
```

Comment:
here, no need for the temp var, just `return sdk.SetPreferredAndroidSdkPath(path);` should be enough